src/button-dropdown/items-list.tsx

```diff
@@ -34,6 +35,7 @@ export default function ItemsList({
           onItemActivate={onItemActivate}
           disabled={item.disabled ?? categoryDisabled}
           highlighted={isHighlighted(item)}
+          focused={isFocused(item)}
```

Comment:
Does this work when an item was highlighted with the keyboard and then user highlights it with the mouse? As the highlighted index remains unchanged and the isFocused is calculated from a ref, the style might not change as expected as no re-render would occur.

Reply:
This is a good point - it shouldn't work. At least for the first time this happens, afterwards the callback might be memoized. It would be good if we could add a test for this case.

Reply:
Good points. Changed the usingMouse as state to update isFocus so that the style would update when changing from keyboard to mouse. Also added a test for this case.